packages/ts/generator-typescript-plugin-model/src/MetadataParser.ts

```diff
@@ -0,0 +1,57 @@
+import type { Schema } from '@hilla/generator-typescript-core/Schema.js';
+import ts, { type ObjectLiteralExpression, type PropertyAssignment } from 'typescript';
+
+interface Annotation {
+  name: string;
+  attributes?: Record<string, unknown>;
+}
+
+export type SchemaWithMetadata = Schema & {
+  'x-annotations'?: Annotation[];
+  'x-java-type'?: string;
+};
+
+export class MetadataParser {
```

Comment:
```suggestion
export class MetadataProcessor {
```
The `parser` parts existed just because we needed to parse a string representation of validation constraints. Now we get rid of it, so let's have it consistent.

Reply:
Done